frontend/pages/admin/IntegrationsPage/cards/ConditionalAccess/ConditionalAccess.tsx

```diff
@@ -242,173 +227,210 @@ const ConditionalAccess = () => {
     },
     onError: () => {
       // distinct from successful confirmation response of `false`, this handles an API error
-      setPhase(Phase.ConfirmationError);
+      setEntraPhase(EntraPhase.ConfirmationError);
     },
   });
 
   const {
-    microsoft_entra_tenant_id: contextConfigMsetId,
-    microsoft_entra_connection_configured: contextConfigMseConfigured,
+    microsoft_entra_tenant_id: entraTenantId,
+    microsoft_entra_connection_configured: entraConfigured,
+    okta_idp_id: oktaIdpId,
   } = contextConfig?.conditional_access || {};
 
-  // only checks if tenant id already present in config, not if user added it to the form
+  // Determine if Okta is configured (all 4 fields must be present)
+  const oktaConfigured = !!(
+    oktaIdpId &&
+    contextConfig?.conditional_access?.okta_assertion_consumer_service_url &&
+    contextConfig?.conditional_access?.okta_audience_uri &&
+    contextConfig?.conditional_access?.okta_certificate
+  );
+
+  // Check if this is a managed cloud deployment (Microsoft Entra requires proxy infrastructure)
+  const isManagedCloud = contextConfig?.license?.managed_cloud || false;
+
+  // Check Entra configuration state
   useEffect(() => {
-    if (contextConfigMsetId) {
-      if (!contextConfigMseConfigured) {
-        setPhase(Phase.ConfirmingConfigured);
+    // Don't check config if we're showing the awaiting OAuth banner
+    if (showAwaitingOAuthBanner) {
+      setEntraPhase(EntraPhase.NotConfigured);
+      return;
```

Comment:
Is the intention of this early return the same as putting the rest of the code in this useEffect
callback in an `else` block? If so, an `else` would be more clear and canonical